presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -296,16 +305,65 @@ public ConnectorSplitSource getSplits(
                 splitSchedulingContext.schedulerUsesHostAddresses(),
                 layout.isPartialAggregationsPushedDown());
 
-        HiveSplitSource splitSource = computeSplitSource(splitSchedulingContext, table, session, hiveSplitLoader);
+        HiveSplitSource splitSource = computeSplitSource(splitSchedulingContext, table, session, hiveSplitLoader, ratio);
         hiveSplitLoader.start(splitSource);
 
         return splitSource;
     }
 
+    private double getSplitScanRatio(
+            ConnectorSession session,
+            SchemaTableName tableName,
+            HiveTableLayoutHandle layout,
+            TransactionalMetadata metadata,
+            List<HivePartition> partitions)
+    {
+        double ratio = 1.0;
+        if (!isDynamicSplitSizesEnabled(session)) {
+            return ratio;
+        }
+        HiveTableHandle hiveTableHandle = new HiveTableHandle(tableName.getSchemaName(), tableName.getTableName());
+
+        Set<HiveColumnHandle> readColumnHandles = new HashSet<>();
+        if (layout.getRequestedColumns().isPresent()) {
+            readColumnHandles.addAll(layout.getRequestedColumns().get());
+        }
+        readColumnHandles.addAll(layout.getPredicateColumns().values());
+        readColumnHandles.removeIf(handle -> handle.isHidden() || handle.isPartitionKey());
+
+        Map<String, Type> readColumnTypes = readColumnHandles.stream().collect(toImmutableMap(HiveColumnHandle::getName, handle -> metadata.getColumnMetadata(session, hiveTableHandle, handle).getType()));
+        Map<String, ColumnHandle> readColumns = readColumnHandles.stream().collect(toImmutableMap(HiveColumnHandle::getName, identity()));
+
+        TableStatistics tableStatistics = metadata.getHiveStatisticsProvider().getTableStatistics(session, tableName, readColumns, readColumnTypes, partitions);
+        double totalSize = tableStatistics.getTotalSize().getValue();
+        double requiredSize = 0;
```

Comment:
what does requiredSize mean? it looks like we're getting the total size of all columns we're selecting.  What's "required" about it?

Reply:
Updated to readSize